compose/service.py

```diff
@@ -1250,7 +1256,7 @@ def pull(self, ignore_pull_failures=False, silent=False, stream=False):
         return progress_stream.get_digest_from_pull(event_stream)
 
     def push(self, ignore_push_failures=False):
-        if 'image' not in self.options or 'build' not in self.options:
+        if not self.can_be_pushed():
```

Comment:
Wondering; should we log something if the service _does_ have an `image:` specified, but either doesn't have `build:` or image has a _digest_ set (to give some clue why the image for a service wasn't pushed)?

Reply:
It would be great to have logs!
For services without `build` and|or `image` just info log, that they were skipped.